init/eessi_environment_variables

```diff
@@ -76,11 +76,17 @@ if [ -d $EESSI_PREFIX ]; then
             false
           fi
 
-          export LMOD_RC="$EESSI_SOFTWARE_PATH/.lmod/lmodrc.lua"
-          if [ -f $LMOD_RC ]; then
-            show_msg "Found Lmod configuration file at $LMOD_RC"
+          export LMOD_CONFIG_DIR="$EESSI_SOFTWARE_PATH/.lmod"
+          if [ -f $LMOD_CONFIG_DIR/lmodrc.lua ]; then
+            show_msg "Found Lmod configuration file at $LMOD_CONFIG_DIR/lmodrc.lua"
           else
-            error "Lmod configuration file not found at $LMOD_RC"
+            error "Lmod configuration file not found at $LMOD_CONFIG_DIR/lmodrc.lua"
+          fi
+          # Allow for site-specific lmod configuration
+          host_lmod_rc="${EESSI_PREFIX/versions/host_injections}"/.lmod/lmodrc.lua
```

Comment:
While I get that there probably is no good reason to have separate user hooks, I think we can do better here. Let's first try `"${LMOD_CONFIG_DIR/versions/host_injections}"/lmodrc.lua` and then fall back to the more global location if that doesn't exist, that way sites can have arch-specific customisations.

Reply:
That's a good idea, we can make it hierarchical ourselves. If the more specific one doesn't exist, we just pick up on the generic one. I'll change it.